fix: handle SystemTransactionApplied events from MidnightSystem pallet

cosmir17 opened the pull request:
## Summary
This PR adds support for capturing system transactions that are generated by the node (not submitted via extrinsics). These transactions are emitted as `SystemTransactionApplied` events by the MidnightSystem pallet.

## Context
- The node can generate system transactions internally (e.g., CNightGeneratesDust)
- These appear as `SystemTransactionApplied` events rather than extrinsics
- Oscar implemented similar handling in the node toolkit (PR #60)
- This functionality exists in our feat/cnight-generates-dust branch but was never in main
- Adding this support ensures the indexer captures ALL transactions, not just user-submitted ones

## Changes
- Modified `chain-indexer/src/infra/subxt_node/runtimes.rs` to handle `SystemTransactionApplied` events
- Made the transactions vector mutable to allow appending system transactions discovered in events
- Added a match arm for `MidnightSystem::SystemTransactionApplied` events in the event processing loop

## Technical Details
The implementation adds handling for `SystemTransactionApplied` events in the `make_block_details` macro:
```rust
Ok(Event::MidnightSystem(midnight_system::Event::SystemTransactionApplied(e))) => {
    // System transactions created by the node (not from extrinsics).
    transactions.push(Transaction::System(ByteVec::from(e.serialized_system_transaction.clone())));
}
```

## Related Issues
- Task: https://shielded.atlassian.net/browse/PM-19760
- Similar implementation: midnight-node PR #60 (node toolkit)
- Cherry-picked from: feat/cnight-generates-dust branch

## Impact
This enhancement ensures the indexer captures all transaction types, including system-generated ones. This is particularly important for features like DUST generation where the system creates transactions automatically.

hseeberger commented on chain-indexer/src/infra/subxt_node/runtimes.rs:
```diff
@@ -167,8 +167,15 @@ macro_rules! make_block_details {
+                        Ok(Event::Session(NewSession { .. })) => {
+                            *authorities = None;
+                        }
+                        Ok(Event::MidnightSystem(midnight_system::Event::SystemTransactionApplied(e))) => {
```
s/e/transaction_applied/

hseeberger commented on chain-indexer/src/infra/subxt_node/runtimes.rs:
```diff
@@ -167,8 +167,15 @@ macro_rules! make_block_details {
+                        }
+                        Ok(Event::MidnightSystem(midnight_system::Event::SystemTransactionApplied(e))) => {
+                            // System transactions created by the node (not from extrinsics).
+                            transactions.push(Transaction::System(ByteVec::from(e.serialized_system_transaction.clone())));
```
This needs formatting (too long line).

cosmir17 commented:
Thanks @hseeberger for the code hygiene improvements and especially for removing those macros!

The explicit functions are much cleaner and Rust Analyzer will be much happier. I agree that when we support multiple protocol versions, the implementations will likely diverge anyway, so the macros weren't providing much value.

Your changes look good:
- Fixed the formatting and naming issues
- Removed the unnecessary clone and type annotation
- Added proper error handling for the events iteration
- Most importantly, replaced macros with explicit functions for better IDE support

Merging now. Thanks for the quick review and improvements!
